From bd60d873270ae0e9317d463cbea10889b90d7317 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 28 Apr 2022 14:43:30 +0800
Subject: [PATCH] u-boot-s32: Aptiv: add u-boot support for Aptiv CVC board

This patch is based on the files fsl-s32g274aaptiv.dts and s32g274aaptiv_defconfig
from the meta-aptiv layer.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm/dts/Makefile              |   1 +
 arch/arm/mach-s32/s32g2/Kconfig    |  15 ++
 board/nxp/aptiv_cvc_sousa/Kconfig  |  15 ++
 board/nxp/aptiv_cvc_sousa/Makefile |   7 +
 configs/aptiv_cvc_sousa_defconfig  |  84 ++++++++
 drivers/net/dwc_eth_qos_s32cc.c    |   1 +
 include/configs/s32g2xxaevb.h      |  14 +-
 8 files changed, 434 insertions(+), 1 deletion(-)
 create mode 100644 board/nxp/aptiv_cvc_sousa/Kconfig
 create mode 100644 board/nxp/aptiv_cvc_sousa/Makefile
 create mode 100644 configs/aptiv_cvc_sousa_defconfig

diff --git a/arch/arm/mach-s32/s32g2/Kconfig b/arch/arm/mach-s32/s32g2/Kconfig
index f6e4733ab5..2f9ea5f5e7 100644
--- a/arch/arm/mach-s32/s32g2/Kconfig
+++ b/arch/arm/mach-s32/s32g2/Kconfig
@@ -20,6 +20,20 @@ config TARGET_S32G274ARDB2
 	  18 CAN ports, I2C and DSPI.
 	select NXP_S32GRDB_BOARD
 
+config TARGET_APTIV_CVC_SOUSA
+	bool "Support for Aptiv CVC board"
+	help
+	  This Aptiv CVC board is based on S32G274A SoC.
+	imply DM_USB
+	imply I2C_EEPROM
+	imply PHY
+	imply PHY_BROADCOM
+	imply SJA1105
+	imply USB
+	imply USB_EHCI_HCD
+	select MISC_INIT_R
+	select NXP_S32GEVB_BOARD
+
 config TARGET_S32G2XXAEVB
 	bool "Support S32G2XXAEVB board"
 	help
@@ -63,6 +77,7 @@ endchoice
 source "board/nxp/s32-cc/s32g/Kconfig"
 source "board/nxp/s32g274ardb2/Kconfig"
 source "board/nxp/s32g2xxaevb/Kconfig"
+source "board/nxp/aptiv_cvc_sousa/Kconfig"
 source "board/nxp/s32g2xxaevb3/Kconfig"
 source "board/nxp/s32g274abluebox3/Kconfig"
 source "board/nxp/s32g274aemu/Kconfig"
diff --git a/board/nxp/aptiv_cvc_sousa/Kconfig b/board/nxp/aptiv_cvc_sousa/Kconfig
new file mode 100644
index 0000000000..4eab69ed98
--- /dev/null
+++ b/board/nxp/aptiv_cvc_sousa/Kconfig
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: GPL-2.0+
+# Copyright 2022 NXP
+
+if TARGET_APTIV_CVC_SOUSA
+
+config SYS_CONFIG_NAME
+	default "s32g2xxaevb"
+
+config SYS_BOARD
+	default "s32g2xxaevb"
+
+config NR_DRAM_BANKS
+	default 2
+
+endif
diff --git a/board/nxp/aptiv_cvc_sousa/Makefile b/board/nxp/aptiv_cvc_sousa/Makefile
new file mode 100644
index 0000000000..9bbb5c34ab
--- /dev/null
+++ b/board/nxp/aptiv_cvc_sousa/Makefile
@@ -0,0 +1,7 @@
+#
+# Copyright 2022 NXP
+#
+# SPDX-License-Identifier:      GPL-2.0+
+#
+
+obj-y += ../s32-cc/s32g/
diff --git a/configs/aptiv_cvc_sousa_defconfig b/configs/aptiv_cvc_sousa_defconfig
new file mode 100644
index 0000000000..3342abb576
--- /dev/null
+++ b/configs/aptiv_cvc_sousa_defconfig
@@ -0,0 +1,79 @@
+CONFIG_ARM=y
+CONFIG_ARM_SMCCC=y
+CONFIG_ARCH_S32G2=y
+CONFIG_SYS_TEXT_BASE=0x340a0000
+CONFIG_ENV_SIZE=0x2000
+CONFIG_ENV_SECT_SIZE=0x2000
+CONFIG_ENV_OFFSET=0x1e0000
+CONFIG_TARGET_APTIV_CVC_SOUSA=y
+CONFIG_S32_LPDDR4=y
+# CONFIG_SYS_MALLOC_F is not set
+CONFIG_TOOLS_DEBUG=y
+CONFIG_FIT=y
+CONFIG_SD_BOOT=y
+CONFIG_USE_BOOTARGS=y
+CONFIG_MD5SUM_VERIFY=y
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_DM=y
+# CONFIG_DM_DEVICE_REMOVE is not set
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MXC=y
+CONFIG_MISC=y
+CONFIG_I2C_EEPROM=y
+CONFIG_SYS_EEPROM_SIZE=128
+CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
+CONFIG_DM_MMC=y
+CONFIG_FSL_USDHC=y
+CONFIG_DM_SPI_FLASH=y
+# CONFIG_SPI_FLASH_BAR is not set
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_BROADCOM=y
+CONFIG_PHY_MICREL=y
+CONFIG_PHY_MICREL_KSZ90X1=y
+CONFIG_PHY_FIXED=y
+CONFIG_DM_ETH=y
+CONFIG_FSL_PFENG=y
+CONFIG_FSL_PFENG_EMAC_1_RGMII=y
+CONFIG_DWC_ETH_QOS_DEVICES=y
+CONFIG_DWC_ETH_QOS_S32CC=y
+CONFIG_RGMII=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_PCIE_S32GEN1=y
+CONFIG_PHY=y
+CONFIG_DM_PMIC=y
+CONFIG_DM_PMIC_VR5510=y
+CONFIG_FSL_LINFLEXUART=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_FSL_DSPI=y
+CONFIG_FSL_QSPI=y
+#CONFIG_SJA1105=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_ULPI_VIEWPORT=y
+CONFIG_USB_ULPI=y
+# CONFIG_EFI_LOADER is not set
+CONFIG_CMD_CPU=y
+CONFIG_CMD_EEPROM=y
+CONFIG_CMD_MD5SUM=y
+CONFIG_CMD_MEMTEST=y
+CONFIG_CMD_ADC=y
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_FUSE=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_SPI=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_TIMER=y
+CONFIG_CMD_PMIC=y
+CONFIG_CMD_SMC=y
+CONFIG_FAT_WRITE=y
+CONFIG_MENU=y
+CONFIG_CMD_BOOTMENU=y
diff --git a/include/configs/s32g2xxaevb.h b/include/configs/s32g2xxaevb.h
index 263a511ca6..d07282808b 100644
--- a/include/configs/s32g2xxaevb.h
+++ b/include/configs/s32g2xxaevb.h
@@ -9,13 +9,25 @@
 
 #define EXTRA_BOOTCOMMAND		PFE_INIT_CMD
 #define EXTRA_BOOT_ARGS			PFE_EXTRA_BOOT_ARGS
-#define FDT_FILE			"s32g2xxa-evb.dtb"
 
+#if defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
+
+#define FDT_FILE			"s32g274a-aptiv.dtb"
+#ifdef CONFIG_FSL_PFENG
+#  define PFENG_MODE "enable,sgmii,none,sgmii"
+#  define PFENG_EMAC "0"
+#endif
+
+#else
+
+#define FDT_FILE			"s32g2xxa-evb.dtb"
 #ifdef CONFIG_FSL_PFENG
 #  define PFENG_MODE "enable,none,rgmii,rgmii"
 #  define PFENG_EMAC "1"
 #endif
 
+#endif
+
 #if defined(CONFIG_USB)
 #  define CONFIG_MXC_USB_PORTSC		PORT_PTS_ULPI
 #  define CONFIG_USB_EHCI_MX6
-- 
2.25.1

